File: src/main/java/com/futurice/testtoys/RelativeRange.java
package com.futurice.txttest;

import scala.math.Numeric;

public class RelativeRange {
	
	private final Number VALUE; 
	
	private final double RANGE;
	
	public RelativeRange(Number v, double range) {
		VALUE = v; 
		RANGE = range;
	}
	
	public boolean equals(Object o) {
		boolean rv = false;
		if (o instanceof String) {
			try {
				double d = Double.parseDouble(((String)o));
				rv = d >= VALUE.doubleValue()*(1/RANGE)
				  && d <= VALUE.doubleValue()*RANGE;
			} catch (NumberFormatException e) {}
		}
		return rv;
	}

	public String toString() {
		return VALUE.toString(); 
	}
}

File: src/main/scala/com/futurice/testtoys/RelativeRange.scala
package com.futurice.testtoys

class RelativeRange(val value: Number, val range: Double) {
  override def equals(o: Any): Boolean = {
    var rv: Boolean = false
    if (o.isInstanceOf[String]) {
      try {
        val d: Double = (o.asInstanceOf[String]).toDouble
        rv = d >= value.doubleValue * (1 / range) && d <= value.doubleValue * range
      }
      catch {
        case e: NumberFormatException => {
        }
      }
    }
    return rv
  }

  override def toString: String = {
    return value.toString
  }
}